### PR TITLE
Add JWT authentication example

### PR DESCRIPTION
**Pull request by dhmlau**
While working on https://github.com/strongloop/loopback-next/pull/5421, I think it might be better to have an example app, so that the tutorial can point to the source code. 

What's left: CLI, add this example to Examples.md, and CODEOWNERS file.

Related to: #5421

## Checklist

👉 [Read and sign the CLA (Contributor License Agreement)](https://cla.strongloop.com/agreements/strongloop/loopback-next) 👈

- [x] `npm test` passes on your machine
- [x] New tests added or existing tests modified to cover all changes
- [x] Code conforms with the [style guide](http://loopback.io/doc/en/contrib/style-guide.html)
- [ ] API Documentation in code was updated
- [ ] Documentation in [/docs/site](../tree/master/docs/site) was updated
- [ ] Affected artifact templates in `packages/cli` were updated
- [ ] Affected example projects in `examples/*` were updated

👉 [Check out how to submit a PR](https://loopback.io/doc/en/lb4/submitting_a_pr.html) 👈

### CODE REVIEW

File: examples/todo-jwt/package.json [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+    "@loopback/http-caching-proxy": "^2.1.2",
+    "@loopback/testlab": "^3.1.2",
+    "@types/lodash": "^4.14.150",
+    "@types/node": "^10.17.21",
+    "@typescript-eslint/eslint-plugin": "^2.30.0",
+    "@typescript-eslint/parser": "^2.30.0",
+    "eslint": "^6.8.0",
+    "eslint-config-prettier": "^6.11.0",
+    "eslint-plugin-eslint-plugin": "^2.2.1",
+    "eslint-plugin-mocha": "^6.3.0",
```
Comment (bajtos) [inline examples/todo-jwt/package.json:66]: I believe these plugins are no longer needed, can you please confirm @raymondfeng

File: examples/todo-jwt/src/__tests__/helpers.ts [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+
+export const aLocation = {
+  address: '1 New Orchard Road, Armonk, 10504',
+  geopoint: <GeoPoint>{y: 41.109653, x: -73.72467},
+  get geostring() {
+    return `${this.geopoint.y},${this.geopoint.x}`;
+  },
+};
+
+export function getProxiedGeoCoderConfig(proxy: HttpCachingProxy) {
```
Comment (bajtos) [inline examples/todo-jwt/src/__tests__/helpers.ts:57]: Integration with Geocoder service has been always troublesome. Does your JWT example really need to include a 3rd party service client? If not, then please remove everything related to geocoding from your example.

Comment (raymondfeng) [reply examples/todo-jwt/src/__tests__/helpers.ts:57]: I second.

File: examples/todo-jwt/CHANGELOG.md [hunk 3]
```diff
@@ -0,0 +1,1509 @@
+# Change Log
+
+All notable changes to this project will be documented in this file.
```
Comment (jannyHou) [inline examples/todo-jwt/CHANGELOG.md:3]: wouldn't this file generate automatically?

**Comment by dhmlau** [comment]
Thanks @jannyHou! I created this PR on your branch. https://github.com/strongloop/loopback-next/pull/5509/

File: examples/todo-jwt/CHANGELOG.md (hunk 3 above)
Comment (dhmlau) [reply examples/todo-jwt/CHANGELOG.md]: will remove. thanks.

File: examples/todo-jwt/package.json [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+    "@types/bcryptjs": "^2.4.2"
+  },
+  "devDependencies": {
+    "@loopback/build": "^5.4.0",
+    "@loopback/eslint-config": "^7.0.0",
+    "@loopback/http-caching-proxy": "^2.1.4",
+    "@loopback/testlab": "^3.1.4",
+    "@types/lodash": "^4.14.150",
+    "@types/node": "^10.17.21",
+    "@typescript-eslint/eslint-plugin": "^2.31.0",
```
[the remark this answers is not in the document]
Comment (dhmlau) [reply examples/todo-jwt/package.json]: i've accepted the changes from your branch which contains this change.

File: examples/todo-jwt/package.json [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+    "@typescript-eslint/parser": "^2.31.0",
+    "eslint": "^7.0.0",
+    "eslint-config-prettier": "^6.11.0",
```
[the remark this answers is not in the document]
Comment (dhmlau) [reply examples/todo-jwt/package.json]: i've accepted the changes from your branch which contains this change.

File: examples/todo-jwt/package.json [hunk 6]
```diff
@@ -0,0 +1,82 @@
...
+    "eslint-plugin-eslint-plugin": "^2.2.1",
```
[the remark this answers is not in the document]
Comment (dhmlau) [reply examples/todo-jwt/package.json]: i've accepted the changes from your branch which contains this change.

File: examples/todo-jwt/README.md [hunk 7]
```diff
@@ -0,0 +1,76 @@
...
+This tutorial demonstrates how to add JWT authentication to the
+[Todo application](https://github.com/strongloop/loopback-next/tree/master/examples/todo).
+
+## Usage
+
+Start the application by running npm start and go to
+http://localhost:3000/explorer. You’ll see the 3 new endpoints under
+`UserController` together with the other endpoints under `TodoController`.
+
+![](https://loopback.io/pages/en/lb4/imgs/auth-tutorial-apiexplorer.png)
```
Comment (jannyHou) [inline examples/todo-jwt/README.md:18]: nitpick: missing diagram name here.

**Pull request merged**